pinot-common/src/main/java/com/linkedin/pinot/common/utils/request/RequestUtils.java

```diff
@@ -206,4 +212,43 @@ public static boolean isFitForStarTreeIndex(SegmentMetadata segmentMetadata, Fil
 
     return true;
   }
+
+  /**
+   * Must be called after a successful check of {@link #isFitForStarTreeIndex}
+   * @return true if no fasthll aggregation or rewriting successful, false otherwise
+   */
+  public static boolean performFastHllRewriting(SegmentMetadata segmentMetadata, BrokerRequest brokerRequest) {
+    List<AggregationInfo> aggregationsInfo = brokerRequest.getAggregationsInfo();
+    StarTreeMetadata starTreeMetadata = segmentMetadata.getStarTreeMetadata();
```

Comment:
performFastHllRewriting should only be called after a successful check of existence of startree. If the user specify a customized generated hll column, we should not rewrite the aggregate column. (Since user generate it, he/she should be knowledge of the column name, and Pinot does not have this information, so no rewriting can be done.)